hw/drivers/sensors/lis2dh12/src/lis2dh12.c

```diff
@@ -1463,7 +1485,75 @@ lis2dh12_sensor_set_trigger_thresh(struct sensor *sensor,
         }
     }
 
+err:
+    return rc;
+}
+
+
+/**
+ * Set the trigger threshold values and enable interrupts
+ *
+ * @param ptr to sensor
+ * @param the Sensor type
+ * @param Sensor type traits
+ *
+ * @return 0 on success, non-zero on failure
+ */
+static int
+lis2dh12_sensor_set_trigger_thresh(struct sensor *sensor,
+                                   sensor_type_t type,
+                                   struct sensor_type_traits *stt)
+{
+    int rc;
+    uint8_t fs;
+    struct sensor_itf *itf;
+    sensor_data_t low_thresh;
+    sensor_data_t high_thresh;
+
+    itf = SENSOR_GET_ITF(sensor);
+
+    if (type != SENSOR_TYPE_ACCELEROMETER) {
+        rc = SYS_EINVAL;
+        goto err;
+    }
+
+    memcpy(&low_thresh, &stt->stt_low_thresh, sizeof(low_thresh));
+    memcpy(&high_thresh, &stt->stt_high_thresh, sizeof(high_thresh));
+
+    rc = lis2dh12_get_full_scale(itf, &fs);
+    if (rc) {
+        goto err;
+    }
+
+    if (fs == LIS2DH12_FS_2G) {
+        fs = 16;
+    } else if (fs == LIS2DH12_FS_4G) {
+        fs = 32;
+    } else if (fs == LIS2DH12_FS_8G) {
+        fs = 62;
+    } else if (fs == LIS2DH12_FS_16G) {
+        fs = 186;
+    } else {
+        rc = SYS_EINVAL;
+        goto err;
+    }
+
+    /* Set low threshold and enable interrupt */
+    rc = lis2dh12_set_low_thresh(itf, low_thresh, fs, stt);
+    if (rc) {
+        hal_gpio_irq_release(itf->si_low_pin);
```

Comment:
can we have hal_gpio_irq_release(itf->si_low_pin); in case of error handler inside lis2dh12_set_low_thresh ?
It can easly be put in "err" label there. And since mentioned function does operation on this irq (init/release) it makes sense to handle release in case of error.

Reply:
Agreed. Will move it in.